### ISSUE TITLE
Option(s) to remove trailing whitespace in multiline strings, comments

### ISSUE THREAD

**Issue by mortenpi** [domluna/JuliaFormatter.jl#784]
JuliaFormatter doesn't remove whitespace in multi-line strings (where it is semantically significant), nor in comments. This is two related, but potentially distinct asks:

- [ ] Removing trailing whitespace in comments. Not entirely sure what the rationale is for not removing it in comments by default? Regardless, feels like minimally there should be an option for it.
- [ ] Opt-in option for removing the semantically significant whitespace in multi-line strings. Theoretically, that whitespace is significant, but in practice it almost never is, and can be stripped (in fact, that might be preferred -- being significant, that whitespace might actually cause bugs). When used in a project that consistently enforce the formatter, this should never cause issues with this changing the program's behavior.

The gripe that motivates this feature request for me is that JuliaFormatter isn't fully consistent with the `Trim Trailing Whitespaces` functionality in VS Code, and it's often quite annoying to fix up the leftover whitespace from other PRs that JuliaFormatter didn't catch.

X-ref: #351

**Comment by domluna** [comment]
yeah this seems reasonable. I'd be open to a PR for it. It shouldn't be that much work to implement

comments could be done in print.jl

for docstrings adding trimming before this should work https://github.com/domluna/JuliaFormatter.jl/blob/master/src/styles/default/pretty.jl#L416-L418

you might have to adjust the offset too but I'm not sure

**Comment by domluna** [comment]
it is works when it might as well be on btw. I don't see why this would be optional assuming it doesn't break anything
